### ISSUE TITLE
Implement RBAC for `svc-bie-kafka` in our `dev` environment. 

### ISSUE THREAD

**Issue by paulsackley-nava** [department-of-veterans-affairs/abd-vro#2530]
Why: Kafka will not be able to connect and receive data. 

**User Story**
As a VRO engineer, I want to update Kafka to use RBAC for authentication and authorization so that we remain compliant. The scope of this ticket is our `dev`, `qa`, and `sandbox` environments. Prod will be handled later, in a separate ticket [#2560].

**Acceptance Criteria**
* Kafka is using RBAC
* Late addition - possibly a new issue - We should ensure that our docs on bie-kafka service have been updated with the RBAC details, as well as any [URL changes](https://github.com/department-of-veterans-affairs/abd-vro/pull/2624#discussion_r1517929249) that we should reflect in the docs.

**Notes about work**
* See documentation here: https://github.com/department-of-veterans-affairs/bip-tooling-confluent/wiki/Streaming-app-authentication-and-authorization-with-Confluent-RBAC
* POCs are Milton Garcia and Darrell Neel
* Slack thread here: https://dsva.slack.com/archives/C04U8GESRT7/p1706727018289089

**Task List**

- [x] 1. Setup up a call with Milton and Megan to talk through blockers 
- [x] 2. Do not continue until we clarify on call what we can do 
- [ ] 3. Set up the QA and Sandbox environments - this is waiting on BIE 
- [x] 4. Set up the integration test to hit the the mock Kafka environments 
- [ ] 5. Test to verify a successful connection and topic subscritpion in the svc-bie-kafka logs of each environment listed above 

hours to complete for QA and Sandbox:  8 hours

**Comment by agile-josiah** [comment]
Who manages our tls certificates after the RBAC changes?
What is the method we are expected to use? (My current understanding is that we are using saslssl/plain)
What are the schema registry urls for each environment?
What is the username/password combo and who maintains this?

**Comment by nelsestu** [comment]
> Who manages our tls certificates after the RBAC changes? What is the method we are expected to use? (My current understanding is that we are using saslssl/plain) What are the schema registry urls for each environment? What is the username/password combo and who maintains this?

@agile-josiah were you able to get answers to these questions or is some of this still unresolved?  If unresolved, I can share how this worked in BIP as that is very likely the answer in this case too.

**Comment by agile-josiah** [comment]
```	at io.confluent.kafka.serializers.AbstractKafkaAvroDeserializer.deserialize(AbstractKafkaAvroDeserializer.java:188) ~[kafka-avro-serializer-7.6.0.jar!/:na]
	at io.confluent.kafka.serializers.KafkaAvroDeserializer.deserialize(KafkaAvroDeserializer.java:108) ~[kafka-avro-serializer-7.6.0.jar!/:na]
	at org.apache.kafka.common.serialization.Deserializer.deserialize(Deserializer.java:73) ~[kafka-clients-7.6.0-ce.jar!/:na]
	at org.apache.kafka.clients.consumer.internals.CompletedFetch.parseRecord(CompletedFetch.java:317) ~[kafka-clients-7.6.0-ce.jar!/:na]
	... 13 common frames omitted
Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.ssl.Alert.createSSLException(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.TransportContext.fatal(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.TransportContext.fatal(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.TransportContext.fatal(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.onConsumeCertificate(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.consume(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.SSLHandshake.consume(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.TransportContext.dispatch(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.SSLTransport.decode(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.SSLSocketImpl.decode(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source) ~[na:na]
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source) ~[na:na]
	at java.base/sun.net.www.protocol.https.HttpsClient.afterConnect(Unknown Source) ~[na:na]
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source) ~[na:na]
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source) ~[na:na]
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source) ~[na:na]
	at java.base/java.net.HttpURLConnection.getResponseCode(Unknown Source) ~[na:na]
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(Unknown Source) ~[na:na]
	at io.confluent.kafka.schemaregistry.client.rest.RestService.sendHttpRequest(RestService.java:313) ~[kafka-schema-registry-client-7.6.0.jar!/:na]
	at io.confluent.kafka.schemaregistry.client.rest.RestService.httpRequest(RestService.java:409) ~[kafka-schema-registry-client-7.6.0.jar!/:na]
	at io.confluent.kafka.schemaregistry.client.rest.RestService.getId(RestService.java:907) ~[kafka-schema-registry-client-7.6.0.jar!/:na]
	at io.confluent.kafka.schemaregistry.client.rest.RestService.getId(RestService.java:880) ~[kafka-schema-registry-client-7.6.0.jar!/:na]
	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.getSchemaByIdFromRegistry(CachedSchemaRegistryClient.java:333) ~[kafka-schema-registry-client-7.6.0.jar!/:na]
	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.getSchemaBySubjectAndId(CachedSchemaRegistryClient.java:464) ~[kafka-schema-registry-client-7.6.0.jar!/:na]
	at io.confluent.kafka.serializers.AbstractKafkaAvroDeserializer$DeserializationContext.schemaFromRegistry(AbstractKafkaAvroDeserializer.java:398) ~[kafka-avro-serializer-7.6.0.jar!/:na]
	... 17 common frames omitted```

**Comment by meganhicks** [comment]
we have established a connection in dev but we are running into certificate areas. We are currently blocked on the cert/SSL error. It infinitely loops.  The BIE team said the timeline below could be resolved by end of week if testing goes okay. <img width="532" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBd015QkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--7aef566f0cfb225c538d7ca7a0668126a749233e/image.png" alt="image.png" />

**Comment by agile-josiah** [comment]
Meeting with Mo today to discuss fixing the connectivity issues.

**Comment by agile-josiah** [comment]
Work to be done: Fixing the integration tests, implementing and verifying higher environments.

**Comment by meganhicks** [comment]
Josiah has requested creds for this from the BIA Team.

**Comment by meganhicks** [comment]
make this a 2nd ticket- Fix mock-bie-kafka to test for the new RBAC/Sasl_ssl auth method

**Comment by meganhicks** [comment]
The latest:


Megan Hicks, PM of VRO
  34 minutes ago
@Darrell Neel
 can I please get those creds in the next few hours? At this point each day we don't get them puts us a day behind on delivering value to our customers.


Mohietosh Gill
  28 minutes ago
@Megan Hicks, PM of VRO
, Sorry for the delay. I sent the credential of IVV environment yesterday only, within few minutes sending UAT environment credential also.


Mohietosh Gill
  10 minutes ago
Credentials for IVV and UAT environment are sent to the consumer.
CC: 
@Megan Hicks, PM of VRO
, 
@Darrell Neel

**Comment by meganhicks** [comment]
break up
